### PR TITLE
feat(tt): surveys

### PR DESCRIPTION
**Pull request by vojtaholik**
This adds survey component driven by our quiz package with some basic displaying logic and context.

https://user-images.githubusercontent.com/25487857/193287765-0cbb1ad6-3cd6-4774-9192-0547f825a013.mp4

ask page screenshot

gif

### CODE REVIEW

File: apps/total-typescript/src/components/app/layout.tsx [hunk 1]
```diff
@@ -70,6 +71,7 @@ const Layout: FunctionComponent<LayoutProps> = ({
         {children}
         {/* {footer ? footer : isNull(footer) ? null : <Footer />} */}
       </div>
+      <SurveyPopup />
```
Comment (vojtaholik) [inline apps/total-typescript/src/components/app/layout.tsx:74]:
```suggestion
      {/* <SurveyPopup /> */}
```

This PR can be merged as is if we comment out this component.

File: packages/quiz/src/hooks/use-question.ts [hunk 2]
```diff
@@ -33,12 +37,15 @@ export default function useQuestion({
   config,
   currentAnswer,
   syntaxHighlighterTheme,
-  questionBodyRenderer,
+  handleSubmitAnswer = defaultSubmitAnswerHandler,
```
Comment (joelhooks) [inline packages/quiz/src/hooks/use-question.ts:40]: needed to be able to override this to do what I wanted without breaking existing use

File: packages/quiz/src/hooks/use-question.ts [hunk 3]
```diff
@@ -50,13 +57,14 @@ export default function useQuestion({
       : currentAnswer
 
   React.useEffect(() => {
-    currentQuestion && send('LOAD_QUESTION', {currentQuestion})
+    currentQuestion &&
+      send('LOAD_QUESTION', {currentQuestion, currentQuestionKey})
```
Comment (joelhooks) [inline packages/quiz/src/hooks/use-question.ts:61]: need to be able to get the key to the question to record with the answer

File: apps/total-typescript/src/hooks/use-survey.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        answerSurveyMutation.mutate({
+          answer,
+          question,
+        })
+      }
+    },
+  })
+
+  React.useEffect(() => {
+    if (subscriber && subscriber.fields) {
```
Comment (vojtaholik) [inline apps/total-typescript/src/hooks/use-survey.tsx:61]: @joelhooks while testing this I noticed that I often ended up not having anything but an `id` in my `ck_subscriber` cookie (no `fields`, etc). would it make sense to call `updateSubscriber` and populate cookie with fresh data more often? e.g. when completing a lesson or subscribing.

**Pull request merged**